### PR TITLE
Update csi-attacher to v4.2.0

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Update csi-attacher to https://github.com/kubernetes-csi/external-attacher/releases/tag/v4.2.0

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

* The rbac file hasn't changed for [3 years](https://github.com/kubernetes-csi/external-attacher/tree/master/deploy/kubernetes).

* Changelog from [3.5.0 to 4.2.0](https://github.com/kubernetes-csi/external-attacher/tree/master/CHANGELOG) shows that the added flag is: --max-grpc-log-length. In external-attacher it is [set to -1](https://github.com/kubernetes-csi/external-attacher/pull/411/files). But in the driver there is a [default 10000](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/commit/f7af69f3665cb77ee5033d17b142a8ab0982577d). We need to set this value at the csi-attacher side.

* The breaking change from 3.x.x to 4.x.x is [--default-fstype](https://github.com/kubernetes-csi/external-attacher/blob/master/CHANGELOG/CHANGELOG-4.0.md#no-really-you-must-read-this-before-you-upgrade), and we need to set that value [here](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/base/controller/controller.yaml#L64). For windows, we've had an explicit storageclass that sets the [fstype to ntfs](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/test/k8s-integration/config/sc-windows.yaml), the default-fstype flag will not do anything for windows.

**Does this PR introduce a user-facing change?**:

```release-note
Update csi-attacher to v4.2.0

```

### CODE REVIEW

File: deploy/kubernetes/base/controller/kustomization.yaml [hunk 1]
```diff
@@ -6,3 +6,10 @@ resources:
 - cluster_setup.yaml
 - controller.yaml
 - v1_csidriver.yaml
+patchesJson6902:
```
Comment (msau42) [inline deploy/kubernetes/base/controller/kustomization.yaml:9]: the patch should only be added to the prow-stable-sidecar-rc-master overlay.

File: deploy/kubernetes/base/controller/max-grpc-log-length.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Set max-grpc-log-length for attacher sidecar.
+- op: add
+  path: /spec/template/spec/containers/0/args/-
```
Comment (msau42) [inline deploy/kubernetes/base/controller/max-grpc-log-length.yaml:3]: csi-attacher is the 2nd container: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/cf727bbf6c79ee91aa77c27220f2b5babc00d1c1/deploy/kubernetes/base/controller/controller.yaml

Comment (sunnylovestiramisu) [reply deploy/kubernetes/base/controller/max-grpc-log-length.yaml]: Updated, and the locally build is:
```
- args:
        - --v=5
        - --csi-address=/csi/csi.sock
        - --http-endpoint=:22012
        - --leader-election
        - --leader-election-namespace=$(PDCSI_NAMESPACE)
        - --timeout=250s
        - --max-grpc-log-length=10000
        env:
        - name: PDCSI_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        image: k8s.gcr.io/sig-storage/csi-attacher:v4.2.0
        livenessProbe:
          failureThreshold: 1
          httpGet:
            path: /healthz/leader-election
            port: http-endpoint
          initialDelaySeconds: 10
          periodSeconds: 20
          timeoutSeconds: 10
        name: csi-attacher
        ports:
        - containerPort: 22012
          name: http-endpoint
          protocol: TCP
        volumeMounts:
        - mountPath: /csi
          name: socket-di
```

**Comment by msau42** [comment]
> The breaking change from 3.x.x to 4.x.x is [--default-fstype](https://github.com/kubernetes-csi/external-attacher/blob/master/CHANGELOG/CHANGELOG-4.0.md#no-really-you-must-read-this-before-you-upgrade), and we set that value [here](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/base/controller/controller.yaml#L38).

That is pointing to csi-provisioner arg, but this is for csi-attacher.

**Comment by sunnylovestiramisu** [comment]
Build it again:

```
  - args:
      - --v=5
      - --csi-address=/csi/csi.sock
      - --http-endpoint=:22012
      - --leader-election
      - --leader-election-namespace=$(PDCSI_NAMESPACE)
      - --timeout=250s
      - --max-grpc-log-length=10000
      - --default-fstype=ext4
      env:
      - name: PDCSI_NAMESPACE
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      image: k8s.gcr.io/sig-storage/csi-attacher:v4.2.0
      livenessProbe:
        failureThreshold: 1
        httpGet:
          path: /healthz/leader-election
          port: http-endpoint
        initialDelaySeconds: 10
        periodSeconds: 20
        timeoutSeconds: 10
      name: csi-attacher
      ports:
      - containerPort: 22012
        name: http-endpoint
        protocol: TCP
      volumeMounts:
      - mountPath: /csi
        name: socket-dir
```

**Comment by sunnylovestiramisu** [comment]
From the log:

```
W0216 17:18:26.167] $ gcloud auth login
E0216 17:18:26.167] Command failed
```

/retest

**Comment by sunnylovestiramisu** [comment]
e2e tests constantly failing with:

```
W0216 19:48:20.799] ERROR: (gcloud.auth.activate-service-account) There was a problem refreshing your current auth tokens: ('invalid_grant: Invalid JWT Signature.', {'error': 'invalid_grant', 'error_description': 'Invalid JWT Signature.'})
```

**Comment by sunnylovestiramisu** [comment]
The issue still exists: https://github.com/kubernetes/test-infra/issues/27157#issuecomment-1220982143

**Comment by sunnylovestiramisu** [comment]
Key has been rotated. 

/retest

**Pull request merged**